web/src/js/__tests__/ducks/modes/localSpec.tsx

```diff
@@ -0,0 +1,147 @@
+import localReducer, {
+    toggleLocal,
+    setApplications,
+    initialState,
+    MODE_LOCAL_TOGGLE,
+    MODE_LOCAL_SET_APPLICATIONS,
+    getMode,
+    sanitizeInput,
+} from "../../../ducks/modes/local";
+import * as options from "../../../ducks/options";
+import { TStore } from "../tutils";
+import fetchMock, { enableFetchMocks } from "jest-fetch-mock";
+
+describe("localReducer", () => {
+    it("should return the initial state", () => {
+        const state = localReducer(undefined, {});
+        expect(state).toEqual(initialState);
+    });
+
+    it("should handle MODE_LOCAL_SET_APPLICATIONS action", () => {
+        const applications = "app1, app2";
+        const action = { type: MODE_LOCAL_SET_APPLICATIONS, applications };
+        const newState = localReducer(initialState, action);
+        expect(newState.applications).toBe(applications);
+        expect(newState.error).toBeUndefined();
+    });
```

Comment:
We don't want to manually construct actions in tests, that's a maintenance nightmare. The test two below covers this comprehensively, this one can go entirely. :)

The only exception to this is actions for which we don't have action creators (OPTIONS_RECEIVE, ...).